WalletWasabi/Services/TransactionProcessor.cs

```diff
@@ -0,0 +1,215 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using WalletWasabi.KeyManagement;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Services
+{
+	public class TransactionProcessor
+	{
+		public KeyManager KeyManager { get; }
+		public ConcurrentHashSet<uint256> TransactionHashes { get; }
+		public ConcurrentHashSet<SmartTransaction> TransactionCache { get; }
+		public ObservableConcurrentHashSet<SmartCoin> Coins { get; }
+		public Money DustThreshold { get; }
+		public event EventHandler<SmartCoin> CoinSpentOrSpenderConfirmed;
+		public event EventHandler<SmartCoin> CoinReceived; 
+
+		public TransactionProcessor(KeyManager keyManager, 
+			ConcurrentHashSet<uint256> transactionHashes, 
+			ObservableConcurrentHashSet<SmartCoin> coins,
+			Money dustThreshold
+		)
+		{
+			KeyManager = keyManager;
+			TransactionHashes = transactionHashes;
+			TransactionCache = new ConcurrentHashSet<SmartTransaction>();
+			Coins = coins;
+			DustThreshold = dustThreshold;
+		}
+
+		public bool Process(SmartTransaction tx)
```

Comment:
Can we rename this method to `ProcessTransaction`

Reply:
No. The class is called `TransactionProcessor` and that means that the `Process` method that received a `SmartTransaction` is there to process a transaction, naming it `ProcessTransaction` is redundant. Look at any method in dotnet, the `SmptClient` for example has a `Send` method and not a `SendEmail` or `SendSmptMessage`. The `Action` class has an `Execute` method instead of an `ExecuteAction` and so on.